resources/patches/scheduler_workbench-allowed_status.patch

```diff
@@ -0,0 +1,173 @@
+-    ),
++  $permissions['override default scheduler time'] = array(
++    'title' => t('Override default scheduler time'),
++    'description' => t('Allow users to override the default un-publish schedule time'),
++  );
++  $permissions['scheduler workbench'] = array(
++    'title' => t('Administer Scheduler Workbench'),
++    'description' => t('Perform administration tasks for Scheduler Workbench.'),
+   );
++  return $permissions;
+ }
+
+ /**
+@@ -141,7 +160,14 @@ function scheduler_workbench_scheduler_node_form_submit($form, $form_state) {
+  */
+ function scheduler_workbench_node_validate($node, $form) {
+   $date_format = variable_get('scheduler_date_format', SCHEDULER_DATE_FORMAT);
+-
++  if (!empty($node->revision_publish_on) || !empty($node->publish_on) ) {
++    // Check if the current status will be published.
++    $allowed_status = variable_get('scheduler_workbench_allowed_status', array());
++    if (!in_array($node->workbench_moderation_state_new, $allowed_status)) {
++      $message = t('The current status @status is not allowed to be published, you will need to update the status.', array('@status' => $node->workbench_moderation_state_new));
++      drupal_set_message($message, 'status', FALSE);
++    }
++  }
+   if (!empty($node->revision_publish_on) && !is_numeric($node->revision_publish_on)) {
+     $publishtime = _scheduler_strtotime($node->revision_publish_on);
+     if ($publishtime === FALSE) {
+@@ -172,7 +198,8 @@ function scheduler_workbench_node_update($node) {
+     $replacements = array(
+       '@publish_time' => format_date($publish_on, 'custom', $date_format),
+     );
+-    drupal_set_message(t('This revision is unpublished and will be published @publish_time.', $replacements), 'status', FALSE);
++    $message = t('This revision is unpublished and will be published @publish_time.', $replacements);
++    drupal_set_message($message, 'status', FALSE);
+   }
+ }
+
+@@ -253,6 +280,23 @@ function _scheduler_workbench_scheduler_unpublish(&$node) {
+  */
+ function _scheduler_workbench_scheduler_pre_publish(&$node) {
+   if ($state = variable_get('scheduler_publish_moderation_state_' . $node->type, FALSE)) {
++    // Get status allowed to be published.
++    $allowed_status = variable_get('scheduler_workbench_allowed_status', array());
++    foreach ($allowed_status as $key => $value) {
++      // If the status is not enabled we remove it from allowed status.
++      if (trim($value) != trim($key)) {
++        unset($allowed_status[$key]);
++      }
++    }
++    // Check if the revision current status allows it to be published.
++    if (!in_array($node->workbench_moderation['my_revision']->state, $allowed_status)) {
++      $options = array(
++        "%node" => $node->nid,
++        '%status' => $node->workbench_moderation['my_revision']->state,
++      );
++      watchdog("scheduler_workbench", t("Node %node can't be published, check status allowed to be published."), $options, WATCHDOG_INFO);
++      return;
```

Comment:
In my tests the watchdog was triggered but the scheduling proceeded as usual so I think the check needs to be triggered somewhere else.